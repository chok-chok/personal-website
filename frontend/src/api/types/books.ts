export interface ListBooksQuery { 
    status?: string
};

export interface Book {
    title: string,
    author: string,
    readStatus: string,
    coverImage: string,
    review?: string,
    rating?: number,
    readDate?: Date,
    id?: string
}

