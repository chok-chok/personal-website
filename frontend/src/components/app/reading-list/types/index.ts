export interface ReadingListHeaderProps {
  headerText: string;
  bodyText: string;
}

export interface RatingProps {
  rating: number;
}

export interface ActivelyReadingBook {
  title: string;
  author: string;
  coverImage: string;
}
