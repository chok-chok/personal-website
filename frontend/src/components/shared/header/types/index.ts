export interface MenuItem {
  title: string;
  url: string;
}

export interface MenuProps {
  menus: MenuItem[];
}
