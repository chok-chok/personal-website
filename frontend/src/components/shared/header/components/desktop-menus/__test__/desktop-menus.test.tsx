import * as React from 'react';
import DesktopMenus from '../desktop-menus';
import { shallow } from 'enzyme';
import { MenuProps } from '../../../types';

const props: MenuProps = {
  menus: [
    {
      title: 'Home',
      url: '/',
    },
    {
      title: 'Reading list',
      url: '/reading-list',
    },
    {
      title: 'Contact',
      url: '/contact',
    },
  ],
};

describe('Test DesktopMenus component', () => {
  it('should renders correctly', () => {
    const tree = shallow(<DesktopMenus menus={props.menus} />);
    
    expect(tree).toMatchSnapshot();
  });

  it('should renders correct number of menus', () => {
    const wrapper = shallow(<DesktopMenus menus={props.menus}/>);

    expect(wrapper.find('.desktop-menu')).toHaveLength(props.menus.length);
  });
});
