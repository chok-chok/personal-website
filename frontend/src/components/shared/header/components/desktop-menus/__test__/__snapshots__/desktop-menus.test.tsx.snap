// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DesktopMenus component DesktopMenus component renders correctly 1`] = `
<Fragment>
  <styled__DesktopMenusWrapper>
    <styled__StyledMenuItem
      href="/"
      key="0"
    >
      Home
    </styled__StyledMenuItem>
    <styled__StyledMenuItem
      href="/reading-list"
      key="1"
    >
      Reading list
    </styled__StyledMenuItem>
    <styled__StyledMenuItem
      href="/contact"
      key="2"
    >
      Contact
    </styled__StyledMenuItem>
  </styled__DesktopMenusWrapper>
</Fragment>
`;
